packages/ra-ui-materialui/src/input/TextInput.tsx

```diff
@@ -25,20 +27,37 @@ export type TextInputProps = InputProps<TextFieldProps> &
  *
  * The object passed as `options` props is passed to the <ResettableTextField> component
  */
-const TextInput: FunctionComponent<TextInputProps> = ({
-    label,
-    format,
-    helperText,
-    onBlur,
-    onFocus,
-    onChange,
-    options,
-    parse,
-    resource,
-    source,
-    validate,
-    ...rest
-}) => {
+
+const textInputStyles: Styles<Theme, TextInputProps> = {
+    root: {
+        '& .MuiOutlinedInput-notchedOutline': {
+            '& legend': {
+                width: ({ label }) =>
+                    label === false || label === '' ? '0.01px' : 'auto',
+            },
+        },
+    },
+};
+
+const useStyles = makeStyles(textInputStyles, { name: 'RaTextInput' });
```

Comment:
Can you put all that at the bottom of the file and put the comment back above the `TextInput` declaration? Thanks!

Reply:
Btw, why did you extract the styles into `textInputStyles` instead of inlining them in the `makeStyles` call as we do everywhere else?

Reply:
Okay!
I refer to ResettableTextField components. I also wondered why it, so I change this code.

Reply:
I fixed it, thank you!